### PR TITLE
First PR 2022.1 adding a new model called mobilenet-yolo-v4-syg

### PR DESCRIPTION
**Pull request by ermubuzhiming**
This PR adds mobilenet-yolo-v4 model trained for vehicles detection on the road

### CODE REVIEW

**Comment by ermubuzhiming** [comment]
Knowing that **OMZ can’t accept models from DarkNet directly (because it’s impossible to determine legal/copyright conditions for the code in DarkNet),  but we can accept reworked/modified models under permissive licenses like MIT, Apache 2.0, BSD 2,3.**
So I wanna make a few extra notes or explaination.
1.The whole mode is bsed on keras and the backbone is replaced by MobileNet not darknet53. 
2. The maximum of detected classes is 9 not 80 classes in COCO nor 20 in PASCAL VOC.
3. We referred to a lot of other people's excellent models, but we thought we should write the original darknet model  in the document. 

Sorry for not realizing that OMZ can’t accept models from DarkNet directly (because it’s impossible to determine legal/copyright conditions for the code in DarkNet). 
If it is ok, I will modify the part about DarkNet in the document.

**Comment by ermubuzhiming** [comment]
Thanks for your reply in e-mail. I would like to explain your reply as follows ：
1. I do not understand the ZBB, but it seems the PR will be dealed after long time.
2.  I compared the model structure in [https://github.com/david8862/keras-YOLOv3-model-set](url) which includes mobilenet + yolov4, It has something same cause the main structure does not changed too much. 
3.  Our team focuses on the application more and modifies the model for the application of several usecase.Such as it is for 4 classes.  It is trained by the dataset which is shoot by camera roadside and made by our team. So it perform well under the condition of camera roadside and this kind of application scenarios is not common too much.
4. Sorry not refers to github repository for the model. I will add it.
5. If it is not suitable that I pull the model more focuing on application to  the 'model' unit in OMZ? It is ok I pull the model in
[https://github.com/openvinotoolkit/open_model_zoo/tree/master/demos](url)?

**Comment by ermubuzhiming** [comment]
I read the file CVS-54371.doc. There are several question I can not solve.
image
I dont know where to find E2E test and  what are the  meaning of DLBenchmark update and OMZ update.

**Comment by vladimir-dudnik** [comment]
@ermubuzhiming Let discuss internal topics through e-mail. Just not change items you do not know what to with

**Comment by vladimir-dudnik** [comment]
according to [CONTRIBUTING.md](https://github.com/openvinotoolkit/open_model_zoo/blob/develop/CONTRIBUTING.md) please re-target PR to `develop` branch

**Comment by vladimir-dudnik** [comment]
@ermubuzhiming please notice there are conflicts

**Comment by eaidova** [comment]
Jenkins ok to test

File: models/public/mobilenet-yolo-syg/mobilenet-yolo-syg.md [hunk 1]
```diff
@@ -0,0 +1,124 @@
+# mobilemet-yolo-syg
+
+## Use Case and High-Level Description
+
+  This is a Keras\* version of `mobilenet-yolov4` model designed to perform real-time vehicle detection
+  on FFmpeg-OpenVINO. It can realize illegal lane change detection and speed detection by developing FFmpeg.
```
Comment (vladimir-dudnik) [inline models/public/mobilenet-yolo-syg/mobilenet-yolo-syg.md:6]: I think saying that model is trained for real-time vehicle detection is enough. Rest of phrase `on FFmpeg-OpenVINO` may confuse people and let them think that model can be used with FFmpeg only.

Could you please also clarify the following phrase: `It can realize illegal lane change detection and speed detection by developing FFmpeg`. What and why should be developed at FFmpeg side in order to detect lane intersection or speed detection?

File: models/public/mobilenet-yolo-syg/mobilenet-yolo-syg.md [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+  The weights are pretrained by BDD100k and retrained by our own dataset.
+  For details see repository <https://github.com/ermubuzhiming/OMZ-files-download/>,
```
Comment (vladimir-dudnik) [inline models/public/mobilenet-yolo-syg/mobilenet-yolo-syg.md:8]: The repository you mention actually does not provide more details on the model. It does not contain model training pipeline or any description of how you actually got your .PB file and how one may fine-tune the model on their specific data.

File: models/public/mobilenet-yolo-syg/mobilenet-yolo-syg.md [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+2. The array of detection summary info, name - `separable_conv2d_30/BiasAdd`,  shape - `1,26,26,27`. The anchor values are `36,75,  76,55,  72,146`.
+
+3. The array of detection summary info, name - `separable_conv2d_38/BiasAdd`,  shape - `1,13,13,27`. The anchor values are `142,110,  192,243,  459,401`.
+
+For each case format is `B,Cx,Cy,N*14,`, where
+    - `B` - batch size
+    - `Cx`, `Cy` - cell index
+    - `N` - number of detection boxes for cell
+
+Detection box has format [`x`,`y`,`h`,`w`,`box_score`,`class_no_1`, ..., `class_no_9`], where:
```
Comment (vladimir-dudnik) [inline models/public/mobilenet-yolo-syg/mobilenet-yolo-syg.md:69]: in dataset description only 4 classes described. What are additional 5 classes?

File: models/public/mobilenet-yolo-syg/mobilenet-yolo-syg.md [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+You can download models and if necessary convert them into Inference Engine format using the Model Downloader and other automation tools as shown in the examples below.
+An example of using the Model Downloader:
+python3 <omz_dir>/tools/downloader/downloader.py --name <model_name>
+An example of using the Model Converter:
+python3 <omz_dir>/tools/downloader/converter.py --name <model_name>
+
+## Legal Information
+
+The original model is distributed under the following
+[license1](https://raw.githubusercontent.com/tensorflow/models/master/LICENSE):
```
Comment (vladimir-dudnik) [inline models/public/mobilenet-yolo-syg/mobilenet-yolo-syg.md:100]: what relation is between your original model and tensorflow models garden? As far as I understand you took yolo-v4 model with mobilenet backend from david's repo and retrained it on your data, is not it?

File: models/public/mobilenet-yolo-syg/model.yml [hunk 5]
```diff
@@ -0,0 +1,46 @@
+# Copyright © 2018-2019 Intel Corporation
```
Comment (vladimir-dudnik) [inline models/public/mobilenet-yolo-syg/model.yml:1]:
```suggestion
# Copyright © 2021 Intel Corporation
```

File: models/public/mobilenet-yolo-syg/model.yml [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+    sha256: 0014370f7236a59cacdbc74bfea83327e0e99f1811f97a1ff6e2b9c70ce67618
+    source: https://github.com/legendary111/mobilenet-yolo-syg/releases/download/mobilenet-yolo-syg/yolo_pb2ir.json
+
+model_optimizer_args:
+  - --input_model=$dl_dir/mobilenet-yolo-syg.pb
+  - --input_shape=[1,416,416,3]
+  - --input=input_1
+  - --scale_values=input_1[255]
+  - --transformations_config=$dl_dir/yolo_pb2ir.json
+  - --data_type=FP16
```
Comment (vladimir-dudnik) [inline models/public/mobilenet-yolo-syg/model.yml:39]:
```suggestion
```
line not needed there

File: models/public/mobilenet-yolo-syg/model.yml [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+  - --reverse_input_channels
+  - --output_dir=$dl_dir/
```
Comment (vladimir-dudnik) [inline models/public/mobilenet-yolo-syg/model.yml:41]:
```suggestion
```
line not needed there

**Review by vladimir-dudnik — CHANGES_REQUESTED** [review]
please review comments

**Comment by legendary111** [comment]
Thank you for your advice. I have modified it and tested it successfully. Please check. However, the openVino CPU device must be specified, otherwise an error will be reported.
0254058eba8942554b652c0f2520f0d
551a480124f106392269afb389bff20

**Comment by vladimir-dudnik** [comment]
> However, the openVino CPU device must be specified, otherwise an error will be reported.

yes, target device have to be specified, but it is better to specify device in accuarcy checker command line argument `-td CPU` rather then hardcode it into model configuration file

**Comment by vladimir-dudnik** [comment]
@legendary111 please also review and fix failed pre-commit check, like this:
06:59:37  data/dataset_definitions.yml:1298: trailing whitespace.

**Comment by legendary111** [comment]
I'm trying to modify it, but I keep making this mistake. How can I solve it? And I didn't find trailing spaces.
0b0541dac80a451e57172ea10c12370
72422464108fe1cee544d094205ce27

**Comment by vladimir-dudnik** [comment]
Jenkins please retry a build

**Comment by vladimir-dudnik** [comment]
> I'm trying to modify it, but I keep making this mistake. How can I solve it? And I didn't find trailing spaces.

I think you need to fix this:
18:58:56  ./models/public/mobilenet-yolo-syg/accuracy-check.yml
18:58:56    96:37     error    no new line character at the end of file  (new-line-at-end-of-file)

comments on trailing whitespaces might be not relevant

@nignatovsky could you please explain why we get comments on trailing whitespaces while it was fixed?

**Comment by vladimir-dudnik** [comment]
@legendary111 comments on whitespaces also might be related to Windows style line ending in your commits. Please refer to git [documentation](https://docs.github.com/en/get-started/getting-started-with-git/configuring-git-to-handle-line-endings) how to configure git to handle line endings automatically

**Comment by legendary111** [comment]
Now Jenkins has passed the test, please check again. Thank you.

**Comment by vladimir-dudnik** [comment]
@legendary111 model quantized and validated with OpenVINO 2021.4 runtime on CPU with AVX2 instruction set

|model|metric|TF|FP32|FP16|FP32-INT8|FP16-INT8|
|-------|-------|---|---|---|---|---|
|mobilenet-yolo-syg|map|84.81|84.53|84.53|77.37|84.38|
|mobilenet-yolo-syg|coco_precision|85.38|84.44|84.43|78.43|84.82|

It seems validation set is too small (only 64 images) and is not representative enough (does not provide required variability in data), so model can't be quantized with good quality. Please extend validation set (having at least 1000 images in validation set would be good, I think).

Also, I'd suggest to change model name to mobilenet-yolo-v4-syg, to directly point to the base model

And do not ignore please comment on your repo content. Having just .PB file in repository, without at least detailed description how did you get it, does not provide community the way to reproduce result or fine-tune the model to their specific task. One of the important goals of Open Model Zoo as a project is to educate of community how OpenVINO can work with variety of different public models, how to convert and infer them with OpenVINO.

File: data/datasets.md [hunk 8]
```diff
@@ -142,6 +142,24 @@ To use this dataset with OMZ tools, make sure `<DATASET_DIR>` contains the follo
...
+
+### Files layout
+* `SYGData0829/dataset_format_VOC2007` - directory containing annotations, images and image sets files directories
+    * `Annotations` - directory containing the SYGData0829 annotation files
+    * `JPEGImages` - directory containing the SYGData0829 images
+    * `ImageSets` - directory containing the SYGData0829 text files specifying lists of images for different tasks
+		* `Main/val.txt` - image sets file for validation of detection tasks
+
+### Datasets in dataset_definitions.yml
+* `SYGData0829` used for evaluation models on SYGData0829 dataset for object detection task. Label map with 4 object categories are used. (model examples: [`mobilenet-yolo-syg`](../models/public/mobilenet-yolo-syg/README.md))
```
Comment (vladimir-dudnik) [inline data/datasets.md:161]:
```suggestion
* `SYGData0829` used for evaluation models on SYGData0829 dataset for object detection task. Label map with 4 object categories are used. (model examples: [`mobilenet-yolo-v4-syg`](../models/public/mobilenet-yolo-v4-syg/README.md))
```
links are tested at precommit, I think this is cause of the precommit fail:
15:50:36  data/datasets.md: URL reference "../models/public/mobilenet-yolo-syg/README.md" target does not exist or is not a file

File: models/public/device_support.md [hunk 9]
```diff
@@ -91,6 +91,7 @@
 | mobilenet-v2-pytorch | YES | YES | YES |
 | mobilenet-v3-large-1.0-224-tf | YES | YES | YES |
 | mobilenet-v3-small-1.0-224-tf | YES | YES | YES |
+| mobilenet-yolo-v4-syg | YES |    |    |
```
Comment (vladimir-dudnik) [inline models/public/device_support.md:94]: I was able to successfully test the model on iGPU, so we can put YES for GPU column (did not try MYRIAD yet)

**Review by vladimir-dudnik — COMMENTED** [review]
aded a couple comments

**Comment by legendary111** [comment]
Yes, I will check again to submit.
But I have two questions. 
First, can  'integral' specify 'Max' instead of '11point' in accuracy-check.yml?
Second, can we not test the accuracy of coco? 1280 Val photos are now used, with an accuracy of less than 1% before and after conversion under map.

**Comment by legendary111** [comment]
I have a question, do MAP and COCO_Precison have to satisfy 1% from FP32 results at the same time?Or are there other criteria?
Looking forward to your reply.

**Comment by legendary111** [comment]
The solutions to the previous three problems are as follows:
First: The validation set was enlarged to 1280 and passed the accuarcy-check test
Second: Change the name to mobilenet-yolo-v4-syg
Third, "How to get a PB model", I uploaded the code and steps to the GitHub project and pointed to the links in Read. me and model.yml.
Please check again. Thank you.

a13020207719daf3878ef36c63c1994

**Comment by vladimir-dudnik** [comment]
tested with updated dataset

|model|metric|TF|FP32|FP16|FP32-INT8|FP16-INT8|
|-------|-------|---|---|---|---|---|
|mobilenet-yolo-syg|map|87.11|86.35|86.35|88.24|87.81|

File: models/public/mobilenet-yolo-v4-syg/README.md [hunk 10]
```diff
@@ -0,0 +1,121 @@
...
+## Accuracy
+
+Accuracy metrics obtained on [SYGDate0829](https://github.com/ermubuzhiming/OMZ-files-download/releases/tag/v1-ly)'SYGDate0829.z01''SYGDate0829.z02''SYGDate0829.z03''SYGDate0829.zip'
+which is our own made\* validation dataset for converted model.
+
+| Metric |  Value |
+| ------ | -------|
+| mAP    | 87.11% |
+
+## Input：
```
Comment (vladimir-dudnik) [inline models/public/mobilenet-yolo-v4-syg/README.md:28]:
```suggestion
## Input
```
no trailing punctuation in heading

**Review by vladimir-dudnik — COMMENTED** [review]
no trailing punctuation in heading

**Comment by legendary111** [comment]
Thank you for your advice.Is there anything that needs to be modified?

**Comment by legendary111** [comment]
Please try again. Thank you

**Comment by vladimir-dudnik** [comment]
@legendary111 thanks, I've also was able to successfully test model with both C++ and Python object_detection_demo using `-at yolo` as model type. Please add model to these demos [models.lst](https://github.com/openvinotoolkit/open_model_zoo/blob/develop/demos/object_detection_demo/python/models.lst) file and list it in demos [readme](https://github.com/openvinotoolkit/open_model_zoo/blob/develop/demos/object_detection_demo/python/README.md) in supported models section.

Also, please add model to these demos tests in [cases.py](https://github.com/openvinotoolkit/open_model_zoo/blob/develop/demos/tests/cases.py)

**Comment by legendary111** [comment]
Looks like the new update from the development branch was wrong?
0e9212397b38e9d88185a1703fb0c34

**Comment by vladimir-dudnik** [comment]
> Looks like the new update from the development branch was wrong?
> 0e9212397b38e9d88185a1703fb0c34

@eaidova FYI

**Comment by vladimir-dudnik** [comment]
@legendary111 we'll fix linter issues in https://github.com/openvinotoolkit/open_model_zoo/pull/2736

File: demos/tests/cases.py [hunk 11]
```diff
@@ -128,7 +128,7 @@ def single_option_cases(key, *args):
     )),
 
     CppDemo(name='gaze_estimation_demo',
-            model_keys=['-m', '-m_fd', '-m_hp', '-m_lm'],
```
Comment (vladimir-dudnik) [inline demos/tests/cases.py:131]: @legendary111 please rebase your branch to actual develop, to not introduce unrelevant changes

**Review by vladimir-dudnik — CHANGES_REQUESTED** [review]

**Comment by legendary111** [comment]
I have synchronized the latest develop project, please check.

**Review by vladimir-dudnik — APPROVED** [review]

**Comment by legendary111** [comment]
Hi, I was wondering how long the merger will take? Is there anything else that needs to be modified?

**Comment by vladimir-dudnik** [comment]
@legendary111 we will merge soon. There is enough time before the nearest public release. No changes expected from your side, thanks.

**Comment by legendary111** [comment]
Thank you. Our team also made a model about lane detection based on ERFNET network. OMZ doesn't have this yet, so we want to contribute this model to OMZ. Therefore, I would like to ask whether this possibility can be pushed up and which aspects need to be paid attention to. After all, it could be a completely new model and detection method.

**Comment by vladimir-dudnik** [comment]
@legendary111 the same requirements apply to each contribution to OMZ. So, model and validation dataset should be under permissive license and available for downloading from third-party resource. We need accuracy checker config file to be able to run validation and either adoption of any existing demo or some new demo to showcase the model
